packages/flame/lib/src/game/flame_game.dart

```diff
@@ -18,52 +19,79 @@ import 'package:meta/meta.dart';
 ///
 /// This is the recommended base class to use for most games made with Flame.
 /// It is based on the Flame Component System (also known as FCS).
-class FlameGame extends ComponentTreeRoot
+class FlameGame<T extends World> extends ComponentTreeRoot
     with Game
     implements ReadOnlySizeProvider {
   FlameGame({
     super.children,
+    T? world,
+    CameraComponent? cameraComponent,
     Camera? camera,
-  }) {
+  })  : _world = world ?? World() as T,
```

Comment:
is it possible to check if T === World where and throw a better exception otherwise? (explaining the issue in more details)

Reply:
Done!